packages/verification-core/src/pdf/pdf.worker.js

```diff
@@ -4,12 +4,30 @@ self.addEventListener('message', async (e) => {
     try {
         PdfWasmWrapper.run(e.data.pdfWasmModule)
 
-        const encryptionKeys = await PdfWasmWrapper.extractEncryptionKeys(e.data.pdfBytes)
+        switch (e.data.cmd) {
+            case 'extract_encryption_keys': {
+                const encryptionKeys = await PdfWasmWrapper.extractEncryptionKeys(e.data.pdfBytes)
 
-        self.postMessage({
-            status: true,
-            encryptionKeys
-        })
+                self.postMessage({
+                    status: true,
+                    encryptionKeys
+                })
+                break
+            }
+
+            case 'decrypt_pdf': {
+                const decryptedPdfBytes = await PdfWasmWrapper.decryptPdf(e.data.pdfBytes, e.data.encryptionKey)
+                if (decryptedPdfBytes === null) {
+                    throw new Error('couldn\t decrypt pdf')
+                }
+
+                self.postMessage({
+                    status: true,
+                    decryptedPdfBytes
+                })
+                break
+            }
```

Comment:
I'd add a default case that throws an error because the cmd doesn't exist

Reply:
Adjusted